attachment_s3/models/ir_attachment.py

```diff
@@ -178,9 +178,37 @@ def _store_file_read(self, fname, bin_size=False):
         else:
             return super(IrAttachment, self)._store_file_read(fname, bin_size)
 
+
+    def _inverse_datas(self):
+        location = self._storage()
+        for attach in self:
+            # compute the fields that depend on datas
+            value = attach.datas
+            bin_data = base64.b64decode(value) if value else b''
+            vals = {
+                'file_size': len(bin_data),
+                'checksum': self._compute_checksum(bin_data),
+                'index_content': self._index(bin_data, attach.datas_fname, attach.mimetype),
+                'store_fname': False,
+                'db_datas': value,
+            }
+            if value and location != 'db':
+                # save it to the filestore
+                self=self.with_context(mimetype=attach.mimetype)
+                vals['store_fname'] = self._file_write(value, vals['checksum'])
+                vals['db_datas'] = False
+
+            # take current location in filestore to possibly garbage-collect it
+            fname = attach.store_fname
+            # write as superuser, as user probably does not have write access
+            super(IrAttachment, attach.sudo()).write(vals)
```

Comment:
:warning: With @sarsurgithub we discovered that this super is there to bypass the write of `ir.attachment` where it pops out the `file_size` and `checksum` fields as it is supposed to be written by this inverse function.

Thus here with this overwrite we loose this bypass.


You need to import the base ir_attachment model:

```
from odoo.addons.base.models.ir_attachment import IrAttachment as BaseIrAttachment
```
Then use the base model for the write.
```suggestion
            super(BaseIrAttachment, attach.sudo()).write(vals)
```